modules/planning/scenarios/traffic_light/unprotected_left_turn/stage_creep.cc

```diff
@@ -59,12 +60,67 @@ Stage::StageStatus TrafficLightUnprotectedLeftTurnStageCreep::Process(
     AERROR << "TrafficLightUnprotectedLeftTurnStageCreep planning error";
   }
 
-  // TODO(all): to be added
+  if (GetContext()->current_traffic_light_overlap_ids.size() == 0) {
+    return FinishScenario();
+  }
+
+  const auto& reference_line_info = frame->reference_line_info().front();
+
+  PathOverlap* traffic_light = nullptr;
+  bool traffic_light_all_green = true;
+  for (const auto& traffic_light_overlap_id :
+       GetContext()->current_traffic_light_overlap_ids) {
+    // get overlap along reference line
+    PathOverlap* current_traffic_light_overlap =
+        scenario::util::GetOverlapOnReferenceLine(reference_line_info,
+                                                  traffic_light_overlap_id,
+                                                  ReferenceLineInfo::SIGNAL);
+    if (!current_traffic_light_overlap) {
+      continue;
+    }
+
+    // set right_of_way_status
+    reference_line_info.SetJunctionRightOfWay(
+        current_traffic_light_overlap->start_s, false);
+
+    auto signal_color = frame->GetSignal(traffic_light_overlap_id).color();
+    ADEBUG << "traffic_light_overlap_id[" << traffic_light_overlap_id
+           << "] start_s[" << current_traffic_light_overlap->start_s
+           << "] color[" << signal_color << "]";
+
+    // check on traffic light color
+    if (signal_color != TrafficLight::GREEN) {
+      traffic_light_all_green = false;
+      break;
+    }
+
+    traffic_light = current_traffic_light_overlap;
+  }
+
+  if (traffic_light == nullptr) {
+    return FinishStage();
+  }
+
+  if (traffic_light_all_green) {
+    // creep
+    // TODO(all): find proper creeping area for left turn
+    const double wait_time =
+        Clock::NowInSeconds() - GetContext()->creep_start_time;
+    const double timeout_sec = scenario_config_.creep_timeout_sec();
+    auto* task =
+        dynamic_cast<CreepDecider*>(FindTask(TaskConfig::CREEP_DECIDER));
+    if (task &&
```

Comment:
"if (task != nullptr ...)" is the preferred style in c++11. "if (task)" does implicit type casting and might not produce the same result if the system changes.